brave/brave-core
# Implement state level ads delivery

Resolves https://github.com/brave/brave-browser/issues/9200

## Submitter Checklist:

- Verified that these changes build without errors on
- Verified that these changes pass automated tests (unit, browser, security tests) on

## Test Plan:
- verify country subdivision fetches only when locale is US (irrespective of VPN setting)
- verify country subdivision retries with exponential backoff if server request fails
- verify country subdivision fetches periodically (in debug ~2 mins other ~24 hours)
- verify subdivision is not fetched if state-level targeting is disabled
- verify that subdivision is not updated after the user manually sets the State
- verify that subdivision targeting is auto-detected by default on a fresh install

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on.
